### PR TITLE
Increasing Test Coverages

### PR DESCRIPTION
**Pull request by iNoles**
I have been trying to increase test coverages in specific modules while doing some cleanup. 

The most challenging part is Jackson and commonMain.

### CODE REVIEW

File: fuel-jackson-jvm/src/main/kotlin/fuel/jackson/ResponseExtension.kt [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
-public inline fun <reified T : Any> HttpResponse.toJackson(mapper: ObjectMapper = defaultMapper): Result<T, Throwable> =
-    doTry(work = {
-        Result.success(mapper.readValue(body.string()))
-    }, errorHandler = {
-        Result.failure(it)
-    })
+public inline fun <reified T : Any> HttpResponse.toJackson(
+    mapper: ObjectMapper = jacksonObjectMapper()
+): Result<T?, Throwable> =
+    runCatching {
```
Comment (kittinunf) [inline fuel-jackson-jvm/src/main/kotlin/fuel/jackson/ResponseExtension.kt:13]: Cool 👍

File: fuel-jackson-jvm/src/test/kotlin/fuel/jackson/FuelJacksonTest.kt [hunk 2]
```diff
@@ -45,19 +51,14 @@ class FuelJacksonTest {
         }
 
         val response = Fuel.get(mockWebServer.url("user-agent").toString())
-        val jackson = response.toJackson<HttpBinUserAgentModel>(createCustomMapper())
+        val jackson = response.toJackson<HttpBinUserAgentModel>(createCustomMapper)
```
Comment (kittinunf) [inline fuel-jackson-jvm/src/test/kotlin/fuel/jackson/FuelJacksonTest.kt:54]: [q] Just wondering whether we could provide a nice DSL for client to configure the mapper?

```kotlin
fun toJackson<T>(configure: ObjectMapper.() -> Unit) {
  val mapper = ObjectMapper()
  mapper.apply(configure)
}
```

**Review by kittinunf — APPROVED** [review]
Not a blocker concern

File: fuel-jackson-jvm/src/test/kotlin/fuel/jackson/FuelJacksonTest.kt (hunk 2 above)
Comment (iNoles) [reply fuel-jackson-jvm/src/test/kotlin/fuel/jackson/FuelJacksonTest.kt:54]: you could have to write a PR about it.

**Pull request merged**